src/gamma_inc.jl

```diff
@@ -426,39 +426,30 @@ See also: [`gamma_inc(a,x,ind)`](@ref SpecialFunctions.gamma_inc)
 """
 function gamma_inc_taylor(a::Float64, x::Float64, ind::Integer)
     acc = acc0[ind + 1]
-    wk = zeros(30)
-    flag = false
-    apn = a + 1.0
-    t = x/apn
-    wk[1] = t
-    loop = 2
-    for indx = 2:20
-        apn += 1.0
-        t *= x/apn
-        if t <= 1.0e-3
-            loop = indx
-            flag = true
-            break
-        end
-        wk[indx] = t
-    end
-    if !flag
-        loop = 20
-    end
-    sm = t
-    tol = 0.5*acc #tolerance
-    while true
+
+    # compute first 21 terms
+    ts = cumprod(ntuple(i -> x / (a + i), Val(21)))
+    last_t = findfirst(<(1.0e-3), ts)
+    last_t = last_t === nothing ? 20 : last_t - 1
+    next_t = last_t + 1
+
+    # sum the smaller terms directly
+    sm = t = ts[next_t]
+    apn = a + next_t
+    tolerance = 0.5acc
+    while t > tolerance
```

Comment:
Maybe we need a do-while here?
Or just copy those lines:
```jl
        apn += 1.0
        t *= x/apn
        sm += t
```

The loop body always runs once.
```jl
    while true
        apn += 1.0
        t *= x/apn
        sm += t
        if t <= tol
            break
        end
    end
```

Reply:
The old one did, this rewrite doesn't need to. The "first loop" is just folded into the initial computation of terms.

Reply:
For a fuller explanation - the original algorithm was:
1. Compute terms one by one (up to 20) and store until they fall below a threshold
2. Compute and sum further terms without storing until it falls below a second threshold
3. Sum that and the previously stored terms in reverse order.

The current one modifies step 1 to:
1. Compute the first 21 terms, keeping the first one falling below the threshold to seed Step 2

This does mean there's some redundancy in computing terms, but the cost of that is far smaller than allocating the array, which is what I was focused on. Actually, let me see if I can eliminate that though, now that I've noticed it...

Reply:
Turns out trying to be precious about re-multiplying some terms is slower, at least for some inputs.